bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/formsportal/Link.java

```diff
@@ -0,0 +1,101 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.forms.core.components.models.formsportal;
+
+import org.osgi.annotation.versioning.ConsumerType;
+
+import com.adobe.cq.wcm.core.components.models.Component;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+/**
+ * Defines the sling model for {@code /apps/core/fd/components/formsportal/link} component.
+ *
+ * @since com.adobe.cq.forms.core.components.models.aemform 1.0.0
+ */
+@ConsumerType
+public interface Link extends Component {
+
+    /**
+     * Defines the Asset Type that is being linked to
+     */
+    enum AssetType {
+        ADAPTIVE_FORM
+    }
+
+    /**
+     * Returns url to which the component is linking to
+     * The URL contains query parameters if configured
+     *
+     * @return URL after processing, or {@code #} if none is set
+     * @since com.adobe.cq.forms.core.components.models.formsportal 1.0.0
+     */
+    default String getUrl() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns the text shown in Link
+     *
+     * @return Title string. It is a mandatory field.
+     * @since com.adobe.cq.forms.core.components.models.formsportal 1.0.0
+     */
+    default String getTitle() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Return the text shown on hovering over link
+     *
+     * @return Tooltip string. Can return null if not set
+     * @since com.adobe.cq.forms.core.components.models.formsportal 1.0.0
+     */
+    default String getTooltip() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Return the strategy with which linked url should be opened.
+     * Refer to {@code <a>} html tag attribute {@code target} for semantics
+     *
+     * @return Target strategy string
+     * @since com.adobe.cq.forms.core.components.models.formsportal 1.0.0
+     */
+    @JsonIgnore
+    default String getTarget() {
```

Comment:
Model should be independent of View, looks like this method is coupling it. Can you remove this ?

Reply:
Okay, will move this logic to htl file

Reply:
Logic moved to a JS use-api alongside component code